src/hdf5_init_meta.c

```diff
@@ -0,0 +1,343 @@
+#include "hdf5_init_meta.h"
+#include "macros.h"
+
+#include <hdf5.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+void create_meta_mq1_fields_dataset(hid_t file, hid_t lcpl, hid_t *meta_handle)
+{
+  if (meta_handle == NULL) {
+    fprintf(stderr, "meta_handle is NULL in create_meta_fields_dataset\n");
+    return;
+  }
+
+  char meta_group_path[64] = "metadata";
+  hid_t meta_group =
+    H5Gcreate(file, meta_group_path, lcpl, H5P_DEFAULT, H5P_DEFAULT);
+  if (meta_group < 0) {
+    fprintf(stderr, "Error creating group in create_meta_mq1_fields_dataset\n");
+    goto cleanup;
+  }
+
+  hsize_t dim[1]       = {0};
+  hsize_t max_dim[1]   = {H5S_UNLIMITED};
+  hsize_t chunk_dim[1] = {1};
+
+  hid_t dataspace = H5Screate_simple(1, dim, max_dim);
+  if (dataspace < 0) {
+    fprintf(stderr,
+            "Error creating dataspace in create_meta_mq1_fields_dataset\n");
+    goto cleanup;
+  }
+
+  hid_t dcpl = H5Pcreate(H5P_DATASET_CREATE);
+  if (dcpl < 0) {
+    fprintf(stderr, "Error creating dcpl in create_meta_mq1_fields_dataset\n");
+    goto cleanup;
+  } else {
+    if (H5Pset_chunk(dcpl, 1, chunk_dim) < 0) {
+      fprintf(stderr,
+              "Error setting chunking in create_meta_mq1_fields_dataset\n");
+      goto cleanup;
+    }
+  }
+
+  hid_t dapl = H5Pcreate(H5P_DATASET_ACCESS);
+  if (dapl < 0) {
+    fprintf(stderr,
+            "Error in creating dapl in create_meta_mq1_fields_dataset\n");
+    goto cleanup;
+  } else {
+  }
+
+  // hid_t header_id_type = H5Tcopy(H5T_C_S1);
+  // H5Tset_size(header_id_type, MQ1_CHAR_LEN_HEADER_ID);
+
+  hid_t pixel_depth_type = H5I_INVALID_HID;
```

Comment:
This is picked up by the gcc static analyser: 

If the above code fails (line 20), it goes to clean-up:
```c
if (meta_group < 0) {
    fprintf(stderr, "Error creating group in create_meta_mq1_fields_dataset\n");
    goto cleanup;
  }
```
And then it goes to the `cleanup` label, where `threshold_type`, `pixel_depth_type` etc will be checked for closing, however they are not defined yet.

I think all the `hid_t pixel_depth_type` etc will need to be declared at the beginning of the function.